buildSrc/src/main/java/GVNightlyVersionVerifier.kt

```diff
@@ -0,0 +1,210 @@
+open class GVNightlyVersionVerifier : Plugin<Project> {
+
+    companion object {
+        private const val GV_VERSION_PATH_FILE = "buildSrc/src/main/java/Gecko.kt"
+        private const val REPO_OWNER = "mozilla-mobile"
+        private const val REPO_NAME = "android-components"
+        private const val BASE_BRANCH_NAME = "master"
+        private const val HEAD = "MickeyMoz"
+    }
+
+    override fun apply(project: Project) {
+
+        project.task("verifyGVNightlyVersion") {
+            setVariables()
+
+            dependsOn("dependencyUpdates")
+
+            doLast {
+                verifyNewGVNightlyVersion(project)
+            }
+        }
+    }
+
+    private fun verifyNewGVNightlyVersion(project: Project) {
+
+        val newVersion = getLastGeckoViewNightlyVersion(project)
+
+        if (newVersion.isNotEmpty()) {
+
+            updateConfigFileWithNewGVNightlyVersion(newVersion)
+
+            val token = project.property("token").toString()
+            val client = GitHubClient(token)
+
+            if (runTests()) {
+
+                if (openAPROnGitHub(client, newVersion))
+                    println("PR Opened Successfully")
+                else
+                    println("Error Creating PR")
+
+            } else {
+                println("Tests failed creating Github Issue")
+                createIssue(client, newVersion)
+            }
+        } else {
+            println("Bad luck :/ no new version of GeckoViewNightly available")
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun getLastGeckoViewNightlyVersion(project: Project): String {
+        val file = File(project.rootDir, "build" + File.separator +
+                "dependencyUpdates" + File.separator + "report.json")
+
+        val json = JsonSlurper().parse(file)
+
+        val outdated = (json as LazyMap)["outdated"]
+        val dependencies = (outdated as Map<*, *>)["dependencies"] as ArrayList<LazyMap>
+
+        val gecko = dependencies.filter {
+            (it["name"] as String).contains("geckoview-nightly")
+        }
+
+        val lastVersion =
+                if (gecko.isNotEmpty())
+                    ((gecko.first()["available"] as LazyMap)["milestone"]) as String
+                else ""
+
+        val actualVersionMajor = nightly_version.split(".")
+        val lastVersionMajor = lastVersion.split(".")
+
+        return if (actualVersionMajor == lastVersionMajor) lastVersion else ""
+    }
+
+    private fun setVariables() {
+        System.setProperty("outputFormatter", "json")
+        System.setProperty("Drevision", "release")
+    }
+
+    private fun updateConfigFileWithNewGVNightlyVersion(newVersion: String) {
+        val file = File(GV_VERSION_PATH_FILE)
+        var fileContent = file.readText()
+        fileContent = fileContent.replace(Regex("nightly_version.*=.*"),
+                "nightly_version = \"$newVersion\"")
+        file.writeText(fileContent)
+        println("${file.name} file updated")
```

Comment:
In a future version it would be nice if this task can write a complete Gecko dependency file. We may end up using this plugin in app projects too and for that it would be nice if this just generates the code file (maybe with [kotlinpoet](https://github.com/square/kotlinpoet)?). Can you file this as follow-up?

Reply:
Sure 👍 , I took a look at kotlinpoet, and I took this approach because the only thing that was channing in the file was the **const nightly_version**